typescript/introduction/tsc-interfaces-classes/tsc-classes-intro.md

```diff
@@ -0,0 +1,77 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Classes Intro
+
+---
+## Content
+
+Classes represent the building blocks of the style of programming called *object-oriented (class-based) programming*, or OOP[1]. 
+
+The idea behind it is to organize related values and behaviors into definitions of entities that exist in your program (an "Animal", a "GamePlayer", a "User", etc). Then you can use these definitions, called classes, to *create (or instantiate)* those entities wherever you'd like. These instances of classes are called objects.
+
+To define a class, TypeScript gives us the `class` keyword. To create an instance of that class we can use the `new` keyword. Let's look at an example:
+
+```ts
+// define an Animal
+class Animal {
+  name: string;
+  constructor(animalName: string) {
+    this.greeting = animalName;
+  }
+  sayHi() {
+    console.log("Hi, I'm a " + this.greeting);
+  }
+}
+
+// create (instantiate) an Animal
+let zebra = new Animal("zebra");
+zebra.sayHi();
+// Hi, I'm a zebra
+```
+
+The benefit of classes is that they allow us to define data and behavior in one place, and then create many entities that use that data and follow that behavior as we see fit. In the example above, we used the `Animal` class to create a `zebra` but we could've also created a `tiger` and an `eagle` too. This allows us to have reusable code and not repeat ourselves.
+
+Note that using classes comes with some drawbacks as well[2]
```

Comment:
let's add a tiny example - "...such as .."

Reply:
The example is in the footnote

Reply:
exactly. without the footnote (which should be optional) the statement doesn't say too much

Reply:
So do you think we should move what's in the footnote here?

Or something like this:

```suggestion
Note that using classes comes with some drawbacks as well due to their strict rules[2]
```